rust/kernel/sync/mod.rs

```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GPL-2.0
+
+mod guard;
+mod mutex;
+
+pub use guard::{Guard, Lock};
+pub use mutex::Mutex;
+
+/// Safely initialises an object that has an `init` function that takes a name and a lock class as
+/// arguments, examples of these are [`Mutex`], [`SpinLock`], and [`Condvar`]. Each of them also
+/// provides a more specialised name that uses this macro.
```

Comment:
Should we provide this if the others will provide the more specialized macro?

Reply:
The name is more specialised but it just calls `init_with_lockdep`. For example, one would call `mutex_init` instead of `init_with_lockdep` though both would work.

Reply:
What I meant is that we could hide `init_with_lockdep!` unless users need that one. The macro needs to be exported because it is called from a user-facing macro, but we could still do `#[doc(hidden)]`.

Reply:
Oh, I see. I've added `#[doc(hidden)]`.